src/expr/src/scalar/func.rs

```diff
@@ -1351,6 +1351,10 @@ fn map_get_values<'a>(a: Datum<'a>, b: Datum<'a>, temp_storage: &'a RowArena) ->
     })
 }
 
+fn map_length(a: Datum) -> Datum {
+    Datum::Int64(a.unwrap_map().iter().count() as i64)
```

Comment:
This is going to be incorrect at the boundary. (I realize it's copied from `list_length`—that one is busted too!) Need to use `try_into`, and produce an IntOutOfRange` error if it fails. 

Btw, I think for consistency with `array_length` we should make these return `i32`. Technically you can create a list or map with more than 2^32 elements, but you're going to have a _very_ bad time if you do. So I think it's okay to change the return type of the length functions to `i32`, and then file a follow up issue to _enforce_ that you can't create an array/map/list with more than 2^32 issues.

Reply:
good point! updated the pr to clean everything up!